src/Metadata/Property/Factory/AnnotationPropertyNameCollectionFactory.php

```diff
@@ -63,7 +63,7 @@ public function create(string $resourceClass, array $options = []): PropertyName
         // Properties
         foreach ($reflectionClass->getProperties() as $reflectionProperty) {
             if (null !== $this->reader->getPropertyAnnotation($reflectionProperty, ApiProperty::class)) {
-                $propertyNames[$reflectionProperty->name] = true;
+                $propertyNames[$reflectionProperty->name] = (string)$reflectionProperty->name;
```

Comment:
Not really because we must ensure that properties are unique. If you use values, you'll need to call `in_array` and it's an O(n) operation instead of O(1).